cdap-ui/app/directives/program-history/program-history.js

```diff
@@ -31,13 +31,24 @@ angular.module(PKG.name + '.commons')
         $scope.programId = $scope.programId || $state.params.programId;
 
         $scope.currentPage = 1;
+
         $scope.$watchCollection('model', function (newVal) {
             if (!angular.isArray(newVal)) {
               return;
             }
             $scope.runs = newVal.map(function (run) {
+              let stateName = $scope.type === 'WORKFLOWS' ? 'workflows.detail.run' : 'spark.detail.run';
               return angular.extend({
-                duration: ( run.end? (run.end - run.start) : 0 )
+                duration: ( run.end? (run.end - run.start) : 0 ),
+                detailUrl: window.getOldCDAPUrl({
```

Comment:
I'm not sure if this is a valid concern or not, but this will break the old realtime pipeline